File: templates/Guilded.NET.Templates.PingUser/CSharp/CommandList.cs
using Guilded.NET;
using Guilded.NET.Util;
using Guilded.NET.Objects;
using Guilded.NET.Objects.Events;
using Guilded.NET.Objects.Chat;
using System.Collections.Generic;
using System.Linq;

namespace ProjectName {
    /// <summary>
    /// List of user bot commands.
    /// </summary>
    public static class CommandList {
        /// <summary>
        /// Responds with `Pong!`
        /// </summary>
        /// <param name="client">Client to post message with</param>
        /// <param name="messageCreated">Message creation event</param>
        /// <param name="command">Name of the command used</param>
        /// <param name="arguments">Command arguments</param>
        [Command("ping", "pong", Description = "Responds with `Pong!`")]
        public static async void Ping(IGuildedClient client, MessageCreatedEvent messageCreated, string command, IList<string> arguments) {
            // Sends a message to channel where `ping`/`pong` command was used
            await client.SendMessageAsync(messageCreated.ChannelId,
                // Generates a new message with content `Pong!`
                Message.Generate("Pong!")
            );
        }
    }
}

File: templates/Guilded.NET.Templates.BasicUser/FSharp/CommandList.fs
namespace ProjectName

open Guilded.NET
open Guilded.NET.Objects.Chat
open Guilded.NET.Objects.Events
/// <summary>
/// List of user bot commands.
/// </summary>
type public CommandList =
    // NOTE: This is how message-based command looks like

    /// <summary>
    /// Responds with `Pong!`.
    /// </summary>
    /// <param name="client">Client that generated a command info and invoked this command method</param>
    /// <param name="info">Information about the command used, with arguments and other things</param>
    /// <param name="messageCreated">Message creation event</param>
    [<Command("ping", "pong", Description = "Responds with `Pong!`")>]
    static member Ping
        (
            client: BasicGuildedClient,
            info: CommandInfo,
            messageCreated: MessageCreatedEvent
        ) =
        // Sends a message to channel where `ping`/`pong` command was used
        "Pong!"
        |> Message.Generate
        |> messageCreated.RespondAsync
        |> Async.AwaitTask
        |> ignore

    // NOTE: This is how forum/comment-based command looks like

    /// <summary>
    /// Responds with `Pong!`.
    /// </summary>
    /// <param name="client">Client that generated a command info and invoked this command method</param>
    /// <param name="info">Information about the command used, with arguments and other things</param>
    /// <param name="replyCreated">Reply/comment creation event</param>
    [<Command("ping", "pong", Description = "Responds with `Pong!`", Placement = CommandPlacement.Comments)>]
    static member PingComment
        (
            client: BasicGuildedClient,
            info: CommandInfo,
            replyCreated: ContentReplyCreatedEvent
        ) =
        // Sends a message to channel where `ping`/`pong` command was used
        "Pong!"
        |> MessageContent.GenerateText // Markdown won't work because of this. Markdown-plain-text is not supported in comments/replies
        |> replyCreated.ReplyAsync
        |> Async.AwaitTask
        |> ignore
